docs/spec.md

```diff
@@ -48,135 +86,154 @@ Here is a WSN defining itself:
                  "(" expression ")" |
                  "{" expression "}" .
     identifier = LETTER { LETTER } .
-    literal    = """ CHARACTER { CHARACTER } """ .
+    literal    = """ CHARACTER { CHARACTER } """ . /* """" is a literal `"` */
     LETTER     = "a" … "z" | "A" … "Z" | "_" .
     CHARACTER  = /* an arbitrary Unicode code point */ .
 
-By convention, upper case production names identify _terminal tokens_.
-Terminal tokens are the leaves in the grammar that cannot be expanded
-further. Lower case production names identify _non-terminals_, which
-are production names that may be expanded further. Lexical tokens are
-enclosed in double quotes `""`. Comments are fenced by `/* … */`.
+_Terminals_ are the leaves in the grammar that cannot be expanded
+further. By convention, terminals are identified by production names
+in uppercase.
+
+_Non-terminals_, on the other hand, can be expanded into other
+productions. This means that they can be replaced by a more complex
+expression. By convention, non-terminals are identified by production
+names in lowercase.
 
-There are two special fencing tokens in evy's grammar related to
+_Literals_ or lexical tokens are enclosed in double quotes `""`.
+Comments are fenced by `/*` … `*/`.
+
+There are two special fencing tokens in Evy's grammar related to
 horizontal whitespace, `<-` … `->` and `<+` … `+>`. `<-` … `->` means
 no horizontal whitespace is allowed between the terminals of the
 enclosed expression, e.g. `3+5` inside `<-` … `->` is allowed, but
 `3 + 5` is not. The fencing tokens `<+` … `+>` are the default and mean
 horizontal whitespace is allowed (again) between terminals.
 
-See section [whitespace](#whitespace) for further details.
+See the section on [whitespace](#whitespace) for further details.
 
-## Evy syntax grammar
+[WSN]: https://en.wikipedia.org/wiki/Wirth_syntax_notation
+[EBNF]: https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form
+[Go Programming Language Specification]: https://go.dev/ref/spec
 
-The `evy` source code is UTF-8 encoded. The NUL character `U+0000` is
-not allowed.
+### Evy syntax grammar
 
-    program    = { statement | func | event_handler | NL } .
-    statements = statement { statement } .
+Evy source code is UTF-8 encoded, which means that it can contain any
+Unicode character. The NUL character `U+0000` is not allowed, as it is
+a special character that is used during compilation.
+
+The `WS` abbreviation in the grammar comments below refers to horizontal
+whitespace, which is any combination of spaces and tabs. The following
+listing contains the complete syntax grammar for Evy.
+
+    program    = { statement | func | event_handler | nl } eof .
+    statements = { nl } statement { statement | nl } .
     statement  = typed_decl_stmt | inferred_decl_stmt |
                  assign_stmt |
                  func_call_stmt |
-                 return_stmt | break_stmt |
```

Comment:
You need to add this `return_stmt | break_stmt ` part back to `statement` production

Reply:
oh sorry! missed it. Fixed!